app/controllers/spree/admin/line_items_controller_decorator.rb

```diff
@@ -19,7 +19,12 @@ def index
 
   def create
     variant = Spree::Variant.find(params[:line_item][:variant_id])
-    OpenFoodNetwork::ScopeVariantToHub.new(@order.distributor).scope(variant)
+    dist = if @order.distributor
+      @order.distributor
+    elsif params[:distributor_id]
+      Enterprise.find(params[:distributor_id])
+    end
+    OpenFoodNetwork::ScopeVariantToHub.new(dist).scope(variant)
```

Comment:
Maybe you can move `dist` to a private method, something similar to:
```ruby
def distributor
  if @order.distributor
    @order.distributor
  elsif params[:distributor_id]
    Enterprise.find(params[:distributor_id])
  end
end
```
BTW, what happens when `dist` is `nil`?